internal/runner/rspec.go

```diff
@@ -63,3 +61,36 @@ func (Rspec) discoveryPattern() DiscoveryPattern {
 		ExcludePattern: excludePattern,
 	}
 }
+
+// commandNameAndArgs returns the command name and arguments to run the Rspec tests
+func (Rspec) commandNameAndArgs(testCases []string) (string, []string) {
```

Comment:
At first read, this implementation was much more complicated than I thought that it would be. Two questions: 

1. Is this slicing necessary? Could we do a straight up string substitution instead? i.e. something like
`cmd = string.ReplaceAll(testCommand, "{{testCases}}", testCases...`

2. If this slicing *is* necessary, I'm finding the naming of the arguments a little confusing. The command seems unnecessarily split into "commandName" and "commandArgs"? Especially when it doesn't really make sense to think of "bundle" as the "commandName" and "exec rspec" as the "commandArgs". I think "cmdPt1" and "cmdPt2" might serve as better names? 

But I think its possible to just pass a single command argument into exec.Command - is there a reason we're not just doing that instead? 

cc @DrJosh9000

Reply:
> Could we do a straight up string substitution instead?

Good call, I think we should be able to simplify the replacement of {{testCases}} with ReplaceAll.

> But also I think its possible to just pass a single command argument into exec.Command - is there a reason we're not just doing that instead

I also thought about this and attempted to pass just one arg `exec.Command(args...)`. It shows error that I'm not providing enough arguments. It looks like I have to follow the definition `func Command(name string, arg ...string) *Cmd`

Reply:
Ahhhh I see, and I also had a more thorough read of the docs for the [command](https://pkg.go.dev/os/exec#Command) function. 

For the command, do we have to be strictly observant of which parts are the "command" and what parts are the "args"? Because strictly speaking, for Rspec the command is "bundle exec" or does it not matter as long as both parts add up to a sensible full command? Just wondering because there's a note in the docs about parsing compatibility with Windows.

In that case, perhaps everything before {{testCases}} is the command? And {{testCases}} and whatever follows is the argument? Which is not too different from your original implementation 😂 Just a case of whether the first word is the command or not 🤔

Reply:
I'm not sure how exec.command works in go. But I tried to use `bundle exec` as the command name and Go isn't able to resolve this path. 
If we have an rspec command `bundle exec rspec test_spec.rb`, I think it needs to look like this for 
 go to understand `exec.Command("bundle", "exec" , "rspec", "test_spec.rb")`

Reply:
Ok sweet thanks for explaining that to me and also running that code 😁